src/mongocrypt-opts-private.h

```diff
@@ -138,6 +143,10 @@ bool _mongocrypt_opts_kms_providers_validate(_mongocrypt_opts_t *opts,
                                              _mongocrypt_opts_kms_providers_t *kms_providers,
                                              mongocrypt_status_t *status) MONGOCRYPT_WARN_UNUSED_RESULT;
 
+// For testing only: register a custom contention factor function with crypt.
```

Comment:
FYI @erwee the API for setting a contention factor is added `mongocrypt-opts-private.h`. The header is not installed, but I expect this API can be used by the server (similar to e.g. [mc_FLE2IndexedEncryptedValueV2_parse](https://github.com/10gen/mongo/blob/d433b42be0531d12228e5705ffe6a4483b14c979/src/mongo/crypto/fle_crypto.cpp#L2970))